src/components/SaveIndicator.tsx

```diff
@@ -0,0 +1,31 @@
+import { useIntl } from "react-intl";
+import { SaveStatus } from "../abstractions/common/save-status";
+
+export function SaveIndicator({ saveStatus }: { saveStatus: SaveStatus }) {
+  const intl = useIntl();
+
+  const { statusClassName, statusText, statusDescription } =
+    saveStatus === "idle"
+      ? {
+          statusClassName: "save-indicator-idle",
+          statusText: intl.formatMessage({ id: "saved" }),
+          statusDescription: intl.formatMessage({ id: "saved_details" }),
+        }
+      : saveStatus === "saved"
+      ? {
+          statusClassName: "save-indicator-saved",
+          statusText: intl.formatMessage({ id: "saved" }),
+          statusDescription: intl.formatMessage({ id: "saved_details" }),
+        }
+      : {
+          statusClassName: "save-indicator-saving",
+          statusText: intl.formatMessage({ id: "saving" }),
+          statusDescription: intl.formatMessage({ id: "saving_details" }),
+        };
+
+  return (
+    <div className={`save-indicator ${statusClassName}`}>
```

Comment:
@GusBaamonde what do you think should be the name of the component class in place of `save-indicator`?